go/pkg/client/client.go

```diff
@@ -474,6 +475,17 @@ func getRPCCreds(ctx context.Context, credFile string, useApplicationDefault boo
 	return rpcCreds, CredsFileAuth, nil
 }
 
+// ResumableWriteOpts sets options required for resumable ByteStream write operations.
+type ResumableWriteOpts struct {
+	FinishWrite       bool
+	LastLogicalOffset int64
```

Comment:
I'm not sure this is what you want to do here. This is an option that applies to the entire client, not just a single request. There can be many individual bytestream requests in flight for the same client.

Reply:
Thanks for the input. Updated the code. 
I removed the field 'ResumableWriteOpts' from Client struct in client.go. I wanted to make the code backward compatible, so I removed changes from the existing method(writeChunked) in bytestream.go and I added new methods taking write options such as offset and a boolean flag for determining to finish a write operation. Please let me know what you think.